src/config.rs

```diff
@@ -2304,6 +2305,40 @@ impl Default for BackupConfig {
     }
 }
 
+#[derive(Clone, Serialize, Deserialize, PartialEq, Debug, OnlineConfig)]
+#[serde(default)]
+#[serde(rename_all = "kebab-case")]
+pub struct BackupStreamConfig {
+    pub num_threads: usize,
+    #[online_config(skip)]
+    pub enable: bool,
+    pub temp_path: String,
+    pub temp_file_size_limit_per_task: ReadableSize,
+}
+
+impl BackupStreamConfig {
+    pub fn validate(&self) -> Result<(), Box<dyn Error>> {
+        if self.num_threads == 0 {
+            return Err("backup.num_threads cannot be 0".into());
+        }
+        Ok(())
+    }
+}
+
+impl Default for BackupStreamConfig {
+    fn default() -> Self {
+        let cpu_num = SysQuota::cpu_cores_quota();
+        Self {
+            // use at most 50% of vCPU by default
+            num_threads: (cpu_num * 0.5).clamp(1.0, 8.0) as usize,
+            enable: false,
+            // TODO: may be use raft store directory
+            temp_path: env::temp_dir().into_os_string().into_string().unwrap(),
```

Comment:
Why not kv dir? Can we leave empty and adjust in `TiKvConfig::validate()`?

Reply:
This files may get leaked (i.e. neither be deleted nor uploaded) when TiKV fails. So leave it at the temporary file dir for get easier for cleaning them. 🤔

Reply:
(BTW, I'm not sure whether make `validate` accepts a `&mut self` would be a good idea...)

Reply:
Temp dir does not sound like a good place, some environment may have a very limited tmp dir, e.g., 1G and even less.